docs/src/design/proposals/APIRedesign_PeteRichards.md

```diff
@@ -0,0 +1,251 @@
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Reducing interface depth (the bundle.json version)](#reducing-interface-depth-the-bundlejson-version)
+  - [Imperitive component registries](#imperitive-component-registries)
+  - [Get rid of "extension category" concept.](#get-rid-of-extension-category-concept)
+  - [Reduce number and depth of extension points](#reduce-number-and-depth-of-extension-points)
+  - [Composite services should not be the default](#composite-services-should-not-be-the-default)
+  - [Get rid of views, representations, and templates.](#get-rid-of-views-representations-and-templates)
+- [Reducing interface depth (The angular discussion)](#reducing-interface-depth-the-angular-discussion)
+  - [More angular: for all services](#more-angular-for-all-services)
+  - [Less angular: only for views](#less-angular-only-for-views)
+- [Standard packaging and build system](#standard-packaging-and-build-system)
+  - [Use systemjs for module loading](#use-systemjs-for-module-loading)
+  - [Use gulp or grunt for standard tooling](#use-gulp-or-grunt-for-standard-tooling)
+  - [Package openmctweb as single versioned file.](#package-openmctweb-as-single-versioned-file)
+- [Misc Improvements](#misc-improvements)
+  - [Refresh on navigation](#refresh-on-navigation)
+  - [Move persistence adapter to promise rejection.](#move-persistence-adapter-to-promise-rejection)
+  - [Remove bulk requests from providers](#remove-bulk-requests-from-providers)
+- [Notes on current API proposals:](#notes-on-current-api-proposals)
+- [[1] Footnote: The angular debacle](#1-footnote-the-angular-debacle)
+  - ["Do or do not, there is no try"](#do-or-do-not-there-is-no-try)
+  - [A lack of commitment](#a-lack-of-commitment)
+  - [Commitment is good!](#commitment-is-good)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
+# Reducing interface depth (the bundle.json version)
+
+## Imperitive component registries 
+
+Transition component registries to javascript, get rid of bundle.json and bundles.json.  Prescribe a method for application configuration, but allow flexibility in how application configuration is defined.
+
+Register components in an imperitive fashion, see angularApp.factory, angularApp.controller, etc.  Alternatively, implement our own application object with new registries and it's own form of registering objects.
+
+## Get rid of "extension category" concept.
+
+The concept of an "extension category" is itself an extraneous concept-- an extra layer of interface depth, an extra thing to learn before you can say "hello world".  Extension points should be clearly supported and documented with whatever interfaces make sense.  Developers who wish to add something that is conceptually equivalent to an extension category can do so directly, in the manner that suites their needs, without us forcing a common method on them.
```

Comment:
I think I agree with this.

What I want to avoid is providing multiple interfaces for the same thing. For instance (assuming the "register during config" approach):

```js
mct.config(['fooProvider', 'barProvider', function (fooProvider, barProvider) {
    fooProvider.registerFoo(myFoo);
    barProvider.bars.push(myBar);
}]);
```

...would be bad (increase number of interfaces needlessly.) We should have/use a common `RegistryProvider.<T>` interface for these (and, ideally, some implementation(s) that can be reused if plugins wish to introduce extension points):

```js
mct.config(['fooProvider', 'barProvider', function (fooProvider, barProvider) {
    fooProvider.register(myFoo);
    barProvider.register(myBar);
}]);
```

This fills the same role as extension categories, so I'm not sure if that would contradict this proposal or not. I agree that we shouldn't force a common method, but we shouldn't prevent using a common method either.